libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -1,19 +1,24 @@
 import { Document, FilterQuery } from 'mongoose';
 import { SoftDeleteModel } from 'mongoose-delete';
-import { BaseRepository } from '../base-repository';
+import { BaseRepository, Omit } from '../base-repository';
 import { IntegrationEntity } from './integration.entity';
 import { Integration } from './integration.schema';
 import { DalException } from '../../shared';
 
-export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
+class PartialIntegrationEntity extends Omit(IntegrationEntity, ['_environmentId', '_organizationId']) {}
+
+type EnforceEnvironmentQuery = FilterQuery<PartialIntegrationEntity & Document> &
+  ({ _environmentId: string } | { _organizationId: string });
```

Comment:
Unfortunately, I could not create a more reusable piece of code. 
I will look into it later and hopefully will come up with more reusable code so we won't have these duplicated lines.

Reply:
```
type EnforceEnvironmentQuery<T> = FilterQuery<T & Document> &
  ({ _environmentId: string } | { _organizationId: string });
```

Wonder if this syntax might be valid.

Reply:
My take here would be to flip things around and create a `PartialIntegrationEntity` with all the content of `IntegrationEntity` but the `_environmentId` and `_organizationId` and then make:
```ts
class IntegrationEntity extends PartialIntegrationEntity {
   _environmentId: string;
   _organizationId: string;
```

So for the `EnforceEnvironmentQuery we can directly use:
```
type EnforceEnvironmentQuery = FilterQuery<PartialIntegrationEntity & Document>
```
What do you think?